source/fuzz/fuzzer_pass_wrap_vector_synonym.cpp

```diff
@@ -0,0 +1,158 @@
+
+        // Get a random vector size from 2 to 4.
+        uint32_t vector_size = GetFuzzerContext()->GetWidthOfWrappingVector();
+
+        // Randomly choose a position that target ids should be placed at.
+        // The position is in range [0, n - 1], where n is the size of the
+        // vector.
+        uint32_t position =
+            GetFuzzerContext()->GetRandomIndexForWrappingVector(vector_size);
+
+        // Target ids are the two scalar ids from the original instruction.
+        uint32_t target_id1 = instruction_iterator->GetSingleWordInOperand(0);
+        uint32_t target_id2 = instruction_iterator->GetSingleWordInOperand(1);
+
+        // Stores the ids of scalar constants.
+        std::vector<uint32_t> vec1_components;
+        std::vector<uint32_t> vec2_components;
+
+        // Width is specified in the first index for either OpTypeInt or
+        // OpTypeFloat.
+        uint32_t width = GetIRContext()
+                             ->get_def_use_mgr()
+                             ->GetDef(operand_type_id)
+                             ->GetSingleWordOperand(0);
+        // Get the scalar type.
+        auto type = GetIRContext()->get_type_mgr()->GetType(operand_type_id);
+        // Whether the constant is signed, not used for float type so default
+        // set to true.
+        bool is_signed_constant =
+            type->AsInteger() ? type->AsInteger()->IsSigned() : true;
+
+        // Populate components based on vector type and size.
+        for (uint32_t i = 0; i < vector_size; ++i) {
+          if (i == position) {
+            vec1_components.emplace_back(target_id1);
+            vec2_components.emplace_back(target_id2);
+          } else {
+            if (type->AsInteger()) {
+              // Operands are integers. Fill other positions with zero
+              // integer constants.
+              vec1_components.emplace_back(FuzzerPass::FindOrCreateZeroConstant(
+                  FuzzerPass::FindOrCreateIntegerType(width,
+                                                      is_signed_constant),
+                  true));
+              vec2_components.emplace_back(FuzzerPass::FindOrCreateZeroConstant(
+                  FuzzerPass::FindOrCreateIntegerType(width,
+                                                      is_signed_constant),
+                  true));
+            } else if (type->AsFloat()) {
+              // Operands are floats. Fill other positions with zero
+              // float constants.
+              vec1_components.emplace_back(FuzzerPass::FindOrCreateZeroConstant(
+                  FuzzerPass::FindOrCreateFloatType(width), true));
+              vec2_components.emplace_back(FuzzerPass::FindOrCreateZeroConstant(
+                  FuzzerPass::FindOrCreateFloatType(width), true));
+            } else {
+              assert(
+                  false &&
+                  "Instruction opcode must be a valid numeric constant type.");
+            }
+          }
```

Comment:
You can replace all this with simply:

```
vec1_components.emplace_back(FindOrCreateZeroConstant(operand_type_id), true));
vec2_components.emplace_back(FindOrCreateZeroConstant(operand_type_id), true));
```

The reason is that `operand_type_id` - the type of the scalar instruction - is exactly the type you're looking for.

Reply:
You are absolutely right. Done.